src/main/scala/edu/berkeley/ce/sparkrocks/Block.scala

```diff
@@ -138,6 +138,116 @@ object Block {
+      return false
+    }
+
+    // Compares points that fall on the x = center._1 line.
+    if ((math.abs(pointA(0) - center(0)) < NumericUtils.EPSILON) &&
+      (math.abs(pointB(0) - center(0)) < NumericUtils.EPSILON)) {
+      // Points furthest away from the center will be before points that are closer to the center
+      if ((pointA(1) - center(1) >= NumericUtils.EPSILON) || (pointB(1) - center(1) >= NumericUtils.EPSILON)) {
+        return pointA(1) > pointB(1)
+      } else {
+        return pointB(1) > pointA(1)
+      }
+    }
+
+    // The cross product of vectors (pointA - center) and (pointB - center) in determinant form. Since it's
+    // in 2-D we're only interested in the sign of the resulting z-vector.
+    val det = (pointA(0) - center(0)) * (pointB(1) - center(1)) -
+      (pointB(0) - center(0)) * (pointA(1) - center(1))
+    // If resulting vector points in positive z-direction, pointA is before pointB
+    if (det > NumericUtils.EPSILON) {
+      true
+    } else if (det < -NumericUtils.EPSILON) {
+      false
+    } else {
+      // pointA and pointB are on the same line from the center, so check which one is closer to the center
+      val d1 = (pointA(0) - center(0)) * (pointA(0) - center(0)) + (pointA(1) - center(1)) * (pointA(1) - center(1))
+      val d2 = (pointB(0) - center(0)) * (pointB(0) - center(0)) + (pointB(1) - center(1)) * (pointB(1) - center(1))
+      d1 > d2
+    }
+  }
+
+  /**
+    * Finds the center of a list of vertices - defined as the average coordinate of all the vertices in the list
+    *
+    * @param vertices List of vertices
+    * @return Coordinate of center of vertices
+    */
+  private def findCenter(vertices: Seq[Array[Double]]): Array[Double] = {
+    Array(vertices.map(_(0)).sum / vertices.length.toDouble,
+      vertices.map(_(1)).sum / vertices.length.toDouble,
+      vertices.map(_(2)).sum / vertices.length.toDouble)
+  }
+
+  /**
+    * Arranges the vertices of each of the faces of the input block such that they are in a counter-
+    * clockwise orientation relative their face's unit normal
+    *
+    * @param block A rock block
+    * @return A mapping from each face of the block to a Seq of vertices for that face arranged in counter-
+    *         clockwise order relative to its unit normal
+    */
+  private def orientVertices(block: Block): Map[Face, Seq[Array[Double]]] = {
+    val faceVertices = block.calcVertices
+    faceVertices.keys.zip(
+      faceVertices.map { case (face, vertices) =>
+        // Rotate vertices to all be in x-y plane
+        val R = Block.rotationMatrix(face.normalVec, Array(0.0, 0.0, 1.0))
+        val rotatedVerts = vertices map { vertex =>
+          val rotatedVertex = R * DenseVector[Double](vertex)
+          Array(rotatedVertex(0), rotatedVertex(1), rotatedVertex(2))
+        }
+        // Order vertices in counter-clockwise orientation
+        val center = Block.findCenter(rotatedVerts)
+        val orderedVerts = {
```

Comment:
I'm not certain, but I think you don't actually need the extra set of braces here. So you can write
```scala
val orderedVerts = if (...) {
    ...
} else {
    ...
}
```